build_library/build_image_util.sh

```diff
@@ -626,26 +644,18 @@ finish_image() {
   local install_grub=0
   local disk_img="${BUILD_DIR}/${image_name}"
 
-  # Copy in packages from the torcx store that are marked as being on disk
-  if [ -n "${FLAGS_torcx_manifest}" ]; then
-    for pkg in $(torcx_manifest::get_pkg_names "${FLAGS_torcx_manifest}"); do
-      local default_version="$(torcx_manifest::default_version "${FLAGS_torcx_manifest}" "${pkg}")"
-      for version in $(torcx_manifest::get_versions "${FLAGS_torcx_manifest}" "${pkg}"); do
-        local on_disk_path="$(torcx_manifest::local_store_path "${FLAGS_torcx_manifest}" "${pkg}" "${version}")"
-        if [[ -n "${on_disk_path}" ]]; then
-          local casDigest="$(torcx_manifest::get_digest "${FLAGS_torcx_manifest}" "${pkg}" "${version}")"
-          sudo cp "${FLAGS_torcx_root}/pkgs/${BOARD}/${pkg}/${casDigest}/${pkg}:${version}.torcx.tgz" \
-            "${root_fs_dir}${on_disk_path}"
-          sudo tar xf "${root_fs_dir}${on_disk_path}" -C "${root_fs_dir}" --wildcards "./usr/share/SLSA"
-          if [[ "${version}" == "${default_version}" ]]; then
-            # Create the default symlink for this package
-            sudo ln -fns "${on_disk_path##*/}" \
-              "${root_fs_dir}/${on_disk_path%/*}/${pkg}:com.coreos.cl.torcx.tgz"
-          fi
-        fi
-      done
-    done
-  fi
+  # Ship the docker systemd-sysext image and rip out torcx in same go; TODO: create seperate sysext images for containerd and docker
+  mkdir -p "${PORTAGE_CONFIGROOT}"/etc/portage/profile
+  query_available_package containerd | sudo_clobber ${PORTAGE_CONFIGROOT}/etc/portage/profile/package.provided # use a temporary package.provided to make emerge believe the dependencies are already installed
+  sudo "${SCRIPTS_DIR}/build_sysext" --board="${BOARD}" --image_builddir=${BUILD_DIR} --squashfs_base="${BUILD_DIR}/${image_sysext_base}" --manglefs_script="${SCRIPTS_DIR}/manglefs_docker" docker-flatcar app-containers/docker
+  sudo install -m 0644 -D "${BUILD_DIR}/docker-flatcar.raw" "${root_fs_dir}"/usr/share/flatcar/
+  sudo mkdir -p "${root_fs_dir}"/etc/extensions/
+  sudo ln -sf /usr/share/flatcar/docker-flatcar.raw "${root_fs_dir}"/etc/extensions/docker-flatcar.raw
+  sudo rm ${PORTAGE_CONFIGROOT}/etc/portage/profile/package.provided
+
+  sudo "${SCRIPTS_DIR}/build_sysext" --board="${BOARD}" --image_builddir=${BUILD_DIR} --squashfs_base="${BUILD_DIR}/${image_sysext_base}" --manglefs_script="${SCRIPTS_DIR}/manglefs_containerd" containerd-flatcar app-containers/containerd
+  sudo install -m 0644 -D "${BUILD_DIR}/containerd-flatcar.raw" "${root_fs_dir}"/usr/share/flatcar/
+  sudo ln -sf /usr/share/flatcar/containerd-flatcar.raw "${root_fs_dir}"/etc/extensions/containerd-flatcar.raw
```

Comment:
@pothos does this work?

Reply:
I also don't quite understand why we first create a docker sysext pretending containerd is installed via `${PORTAGE_CONFIGROOT}/etc/portage/profile/package.provided` and then create a containerd sysext anyway.

Reply:
That was done to have a split containerd and Docker sysext setup. This way users can disable Docker but keep our containerd.

Reply:
(The pretending of containerd being installed is to build the Docker sysext without containerd in it.)

Reply:
@jepio to answer your question this works because of https://github.com/flatcar/scripts/blob/7c255452fac3e6ef67c076a642d0aa08887cf3e4/build_library/build_image_util.sh#L793-L803.

Reply:
> That was done to have a split containerd and Docker sysext setup. This way users can disable Docker but keep our containerd.

Oh I understand the intent, but I'd argue that if we want stackable / dependent sysexts then we should make the packages shipped with each layer explicit, and re-use these when we build the next "upper" layer.

I've extended (build_sysext.sh)[https://github.com/flatcar/scripts/blob/contrib/torcx-deprecation-docker-sysext/build_sysext#L30-L33] to cover this, and updated (build_image_util.sh)[https://github.com/flatcar/scripts/blob/contrib/torcx-deprecation-docker-sysext/build_library/build_image_util.sh#L640-L655] to use it.

Reply:
That looks more complex than it needs to be. How about just mount the whole previous sysext when building the next one.

Reply:
Sysexts do not contain package information since they only cover `/usr` (and/or `/opt`). Installed packages are registered in sub-directories below `/var/db/pkg`.
(I _wish_ this was easier though!)